### PR TITLE
feat: update wallet client signer for local accounts

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

### Focus of this PR:
This PR focuses on improving the WalletClientSigner and TurnkeySigner classes by adding a default account and fixing the signMessage and signTypedData methods.

### Detailed summary:
- Added a default account in WalletClientSigner and TurnkeySigner classes.
- Fixed the signMessage method in WalletClientSigner to use the default account if available.
- Fixed the signTypedData method in WalletClientSigner to use the default account if available.
- Updated the tests for WalletClientSigner and TurnkeySigner classes.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* main
  * **PR #320** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/320?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/320?utm_source=stack-comment).

File: packages/core/src/signer/wallet-client.ts [hunk 1]
```diff
@@ -28,14 +28,16 @@ export class WalletClientSigner implements SmartAccountSigner<WalletClient> {
   readonly signMessage: (
```
Comment (moldy530) [inline packages/core/src/signer/wallet-client.ts:28]: you also need to update the `signTypedData` to do the same thing

File: packages/core/src/signer/wallet-client.ts [hunk 2]
```diff
@@ -28,14 +28,16 @@ export class WalletClientSigner implements SmartAccountSigner<WalletClient> {
...
     message: string | Hex | ByteArray
   ) => Promise<`0x${string}`> = async (message) => {
+    const account = this.inner.account ?? (await this.getAddress());
+    console.log(account)
```
Comment (moldy530) [inline packages/core/src/signer/wallet-client.ts]:
```suggestion
```

File: packages/signers/src/turnkey/__tests__/signer.test.ts [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
+import { TurnkeySigner } from "../signer.js";
 import { TurnkeySubOrganization } from "../types.js";
 
+vi.mock("@turnkey/viem", () => {
+  const createAccount = vi
+    .fn()
+    .mockResolvedValue("0x1234567890123456789012345678901234567890");
+
+  return { createAccount };
+});
```
Comment (moldy530) [inline packages/signers/src/turnkey/__tests__/signer.test.ts:13]: Actually, could we instead mock the `WebAuthnStamper` to return a specific value? that way we can still be sure to test the full flow

**Review by moldy530 — APPROVED** [review]

**Pull request merged**